titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/Transaction.java

```diff
@@ -0,0 +1,17 @@
+package com.netflix.titus.common.framework.reconciler.internal;
+
+import java.util.Optional;
+
+abstract class Transaction {
```

Comment:
any reason this is an abstract class and not an interface?

Reply:
is it to ensure methods themselves are not public?

Reply:
Initially I planned to have common skeleton implementation, but it ended up as a plain interface. I will change that.